src/app/components/client/PetitionBanner.tsx

```diff
@@ -44,6 +51,14 @@ export const PetitionBanner = (props: { user: Session["user"] }) => {
           <TelemetryButton
             variant="primary"
             className={styles.signButton}
+            onPress={() => {
+              // In order to prevent the banner from being hidden before
+              // the click on the link button is getting registered: Delay the
+              // dismissal until the next event loop execution.
```

Comment:
Great comment! (And I presume that that's the reason for the `waitFor` I called out :P)

Reply:
Yup! That’s the reason.